### PR TITLE
[BPK-4177] Replace BpkMapMarker with BpkIconMarker

### PR DESCRIPTION
**Pull request by shaundon**
- `BpkMapMarker` is now `BpkIconMarker`
- I aded a new internal component (`BpkIconMarkerBackground`) to handle the SVG background stuff
- Tested in Safari, Firefox, Edge and IE11, all good
- It has Dima's stamp of approval

screen_shot_2020-09-11_at_10 49 18

Proof of theming working: 
Screen Shot 2020-09-10 at 17 37 11

### CODE REVIEW

**Comment by k0nserv** [comment]
Doesn't it look a bit too tight with some icons, namely the fork and knife one?

File: packages/bpk-component-map/docs/migrating-from-v3-to-v4.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Migrating from `v3` to `v4`
```
Comment (k0nserv) [inline packages/bpk-component-map/docs/migrating-from-v3-to-v4.md:1]: This whole file 👏

**Comment by shaundon** [comment]
> Doesn't it look a bit too tight with some icons, namely the fork and knife one?

You're right, I noticed that too. This is a problem with the icon though, because the layout/spacing in this component matches Figma exactly. I suspect the icon intrudes into the bounding box slightly, whereas other icons have a couple of px of built-in padding.

Maybe I should just use a different icon here to avoid confusion?

**Comment by k0nserv** [comment]
> Maybe I should just use a different icon here to avoid confusion?

That wouldn't solve the issue in prod though, maybe ask Dima what he thinks?

File: packages/bpk-component-map/docs/migrating-from-v3-to-v4.md (hunk 1 above)
Comment (shaundon) [reply packages/bpk-component-map/docs/migrating-from-v3-to-v4.md:1]: Thanks 😄 I based it off of one you wrote a while ago for another component.

**Comment by shaundon** [comment]
> maybe ask Dima what he thinks?

Will do!

**Comment by shaundon** [comment]
Screen Shot 2020-09-11 at 09 43 19

Actually looks alright in Figma, so I'll investigate some more. Maybe there's something wrong with our implementation of that icon in code.

**Review by olliecurtis — COMMENTED** [review]
Looks good to me just going to test this locally

**Comment by shaundon** [comment]
> Looks good to me just going to test this locally

Cool, thanks! I've started doing a 'self review' of my PRs after I make them but before I ask for reviews, it helps me find and fix a lot of problems before the reviewers get to it. Saves a lot of time.

**Review by olliecurtis — APPROVED** [review]

**Pull request merged**